crates/storage/codecs/derive/src/compact/generator.rs

```diff
@@ -55,20 +55,25 @@ pub fn generate_from_to(
     } else {
         quote! {
             #[cfg(test)]
-            #[allow(dead_code)]
-            #[test_fuzz::test_fuzz]
-            fn #fuzz(obj: #ident)  {
-                use #reth_codecs::Compact;
-                let mut buf = vec![];
-                let len = obj.clone().to_compact(&mut buf);
-                let (same_obj, buf) = #ident::from_compact(buf.as_ref(), len);
-                assert_eq!(obj, same_obj);
-            }
+            mod #test {
+                use super::*;
+                use #reth_codecs::__private::test_fuzz;
```

Comment:
nice

Reply:
this is actually a bit problematic because test-fuzz is licensed under AGPL3 and we can't use it unless we use same license as well

right now it works because this is just tests so it only needs to be present in `dev-dependencies` and we can't add it ot `reth_codecs::__private`